test/test_helper.rb

```diff
@@ -144,6 +144,20 @@ def assert_event(tag, expected_additional, actual)
     assert_equal actual.additional_type.new(user_agent_info:, **expected_additional), actual.additional
   end
 
+  # Asserts that both are the same nested hash, using equality checks unless
+  # the value is a hash, in which case it will recursively check the nested hash
+  # The purpose of this assertion is mostly to provide a more useful error when
+  # a nested hash contains different elements instead of overwhelming the user
+  # with a giant nested hash diff that is mostly sorting problems.
```

Comment:
:thinking: isn't Hash not sorted by design?

Reply:
Yes, the problem is when you compare the hashes. If they are different by one key you get a huge diff of all the keys that are out of order between the expected and actual with the 1 key that is wrong hidden somewhere, not even necessarily on lines that are close together.

Reply:
I went ahead and changed the assertion to sort everything, then compare as normal. It dups everything but I think that's fine considering how much easier the output is to read. See comment below.